<launch>
    <include file="$(find tello_sdk)/launch/detect_tags.launch" />
    <arg name="namespace"     default="drone0" />
    <group ns="$(arg namespace)">
        <node name="tello_node" pkg="tello_sdk" type="tello_node"  output="screen">
            <param name="~camera_info_url" value="package://tello_sdk/config/calibration_tello.yaml"/>
            <param name="~camera_name" value="sensor_measurement"/>
        </node>

        <node name="tello_aerostack_node" pkg="tello_sdk" type="tello_aerostack_node"  output="screen">
        </node>

        <node name="tello_tf_broadcaster_node" pkg="tello_sdk" type="tello_tf_broadcaster_node"  output="screen">
        </node>

       <node name="tello_teleop_node" pkg="tello_sdk" type="tello_teleop_node"  output="screen">
       </node>

        <node pkg="nodelet" type="nodelet" args="standalone image_proc/rectify" name="camera" ns="sensor_measurement" output="screen">
            <!--remap rectify inputs-->
            <remap from="image_mono" to="camera/image_raw"/>
            <remap from="camera_info" to="camera/camera_info"/>
            <!--remap rectify outputs-->
            <remap from="image_rect" to="camera/image_rect"/>
        </node>

    </group>

    <group ns="$(arg namespace)/sensor_measurement/camera/image_rect" >
        <rosparam param="disable_pub_plugins">
            - 'image_transport/compressedDepth'
            - 'image_transport/theora'
            - 'image_transport/compressed'
        </rosparam>
    </group>

    <group ns="$(arg namespace)/sensor_measurement/camera" >
        <rosparam param="disable_pub_plugins">
            - 'image_proc/parameter'
        </rosparam>
    </group>

</launch>
